final-project/scripts/main.js

```diff
@@ -8,3 +8,156 @@ const after = before
+    page.target.classList.toggle('page-controller__indicator--active');
+  }
+  
+  const activatePager = (numPages)  => {
+    let carrousel= selected;
+    let carrouselControls = carrousel.querySelector('.carrousel__controls');
+    for (let i = 0; i < numPages; i++) {
+      let pageNode =  document.createElement('span');
+      let addClass = i == 0 ? 'page-controller__indicator page-controller__indicator--active' : 'page-controller__indicator ';
+      pageNode.classList= addClass ;
+      pageNode.addEventListener("click", currentPage); 
+      pageNode.dataset.index = i;
+      carrouselControls.appendChild(pageNode);
+    }
+  }
+
+  const htmlPart = (section,element) => {
+    switch (section) {
+      case "blog":
+        return `
+        <div class="blog__content__article__image">
+          <picture>
+              <source srcset="${element.images.desktop}"
+              media="(min-width:992px)">
+              <source srcset="${element.images.tablet}"
+              media="(min-width:768px)">
+              <img src="${element.images.mobile}" alt="upload" >
+          </picture>
+        </div>
+        <div class="blog__content__article__info">
+          <div  class="blog__content__article__data">
+            <header>
+              <h1>${element.title}</h1>
+            </header>
+            <p>
+              ${element.description}  
+            </p>
+          </div>
+          <div class="blog__content__article__actions">
+              <a href="${element.url}" target="_blank" class="blog__content__article__actions__read" > Read Now </a>
+              <button class="blog__content__article__actions__bookmark" >Add to your bookmarks</button>
+            </div>
+        </div>
+      `
+      case "testimonial":
+        return `
+          <div class="testimonials__gallery__item__quote">
+            <blockquote>
+              ${element.quote}
+            </blockquote>
+          </div>
+          <div class="testimonials__gallery__item__photo">
+              <picture >
+                  <img src="${element.image}" alt="Happy user photo">
+              </picture>
+          </div>
+      `
+      case "customer":
+        return `
+        <span class="customers__list__item__${element.name} ">${element.name}</span>
+      `
+      default:
+        break;
+    }
+  }
+
+  return {loadCarrouselData}
+
+}
+
+let customerClass = "customers__list__item"; 
+let customerTag = 'li'
+let customerCarrouselElement = document.querySelector('.customers.carrousel');
+let customerCarrousel = Carrousel('customer',customerCarrouselElement,customerTag,customerClass,"../data/customers.json");
+customerCarrousel.loadCarrouselData();
+
+let blogCarrouselElement = document.querySelector('.blog.carrousel');
+let itemTag = 'article';
+let itemTagClass = 'blog__content__article';
+let blogCarrousel = Carrousel('blog',blogCarrouselElement,itemTag,itemTagClass,"https://wt-4662f45b9eefda7172b747b28d23efdb-0.sandbox.auth0-extend.com/blog");
```

Comment:
Is nice to have reused the carousel class for testimonials, although the arrow functionality is missing